support/catalogs/images_test.go

```diff
@@ -288,24 +351,81 @@ func TestImageLookupCacheKeyFn(t *testing.T) {
+		pullSecret         []byte
+	}{
+		{
+			name:           "Guest cluster should return true without checking image",
+			olmcatalog:     hyperv1.GuestOLMCatalogPlacement,
+			expectedExists: true,
+			expectedError:  false,
+			pullSecret:     []byte("12345"),
+		},
+		{
+			name:           "Management cluster should fail when image not found",
+			olmcatalog:     hyperv1.ManagementOLMCatalogPlacement,
+			expectedExists: false,
+			expectedError:  true,
+			pullSecret:     []byte("12345"),
+		},
+		{
+			name:               "Management cluster with manifest unknown error should return false",
+			olmcatalog:         hyperv1.ManagementOLMCatalogPlacement,
+			expectedExists:     false,
+			expectedError:      false,
+			imageMetadataError: errors.New("manifest unknown"),
+			pullSecret:         []byte("12345"),
+		},
+		{
+			name:               "Management cluster with unauthorized error should return false",
+			olmcatalog:         hyperv1.ManagementOLMCatalogPlacement,
+			expectedExists:     false,
+			expectedError:      false,
+			imageMetadataError: errors.New("access to the requested resource is not authorized"),
+			pullSecret:         []byte("12345"),
+		},
+		{
+			name:           "Management cluster with successful image check should return true",
+			olmcatalog:     hyperv1.ManagementOLMCatalogPlacement,
+			expectedExists: true,
+			expectedError:  false,
+			pullSecret:     []byte("{\"auths\":{}}"),
+		},
 	}
 
-	// Test to ignore that images don't need to exist for "guest" clusters.
-	fn1 := imageExistsFn(ctx, &hyperv1.HostedControlPlane{Spec: hyperv1.HostedControlPlaneSpec{OLMCatalogPlacement: "guest"}}, pullSecret, fakeMetadataProvider)
-	exists, err := fn1("something")
-	g.Expect(err).ToNot(HaveOccurred())
-	g.Expect(exists).To(Equal(true))
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewGomegaWithT(t)
+			hcp := &hyperv1.HostedControlPlane{
+				Spec: hyperv1.HostedControlPlaneSpec{
+					OLMCatalogPlacement: tc.olmcatalog,
+				},
+			}
+
+			fakeMetadataProvider := &testImageMetadataProvider{
+				FakeRegistryClientImageMetadataProvider: &fakeimagemetadataprovider.FakeRegistryClientImageMetadataProvider{
+					Result:   &dockerv1client.DockerImageConfig{},
+					Manifest: fakeimagemetadataprovider.FakeManifest{},
+				},
+				err: tc.imageMetadataError,
+			}
+
+			fn := imageExistsFn(ctx, hcp, tc.pullSecret, fakeMetadataProvider)
+			exists, err := fn("test-image")
 
-	// Test to ensure we fail if not a guest cluster and can't find image.
-	fn2 := imageExistsFn(ctx, &hyperv1.HostedControlPlane{Spec: hyperv1.HostedControlPlaneSpec{OLMCatalogPlacement: "management"}}, pullSecret, fakeMetadataProvider)
-	exists, err = fn2("something")
-	g.Expect(err).To(HaveOccurred())
-	g.Expect(exists).To(Equal(false))
+			if tc.expectedError {
+				g.Expect(err).To(HaveOccurred())
+			} else {
+				g.Expect(err).ToNot(HaveOccurred())
+			}
+			g.Expect(exists).To(Equal(tc.expectedExists))
```

Comment:
@jparrill do we expect to return a correct value even when we return an `error`? I would say _no_. So I wouldn't check if `exists` is eqaul to `tc.expectedExists`.

Reply:
makes sense. Fixed